pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -277,7 +277,7 @@ func createIngress(isvc *v1beta1.InferenceService, useDefault bool, config *v1be
 		return nil
 	}
 
-	if !isvc.Status.IsConditionReady(v1beta1.PredictorReady) {
+	if isvc.Status.IsConditionFalse(v1beta1.PredictorReady) {
```

Comment:
I think we need to check `IsConditionUnknown` as well to return early, the ingress should not be created until all of the components are in ready state.
https://github.com/kserve/kserve/actions/runs/5215454086/jobs/9413002639?pr=2977